export/mlab_export.py

```diff
@@ -0,0 +1,591 @@
+    logging.info("rrdfile: %s", filename.replace(options.rrddir_prefix, ''))
+  if options.show_rawmetric:
+    logging.info("metric_raw: %s", metric_raw)
+  if options.show_skipped and not metric:
+    logging.info("metric_skipped: %s", metric_raw)
+  if options.show_metric and metric:
+    logging.info("metric: %s", metric)
+
+  return (HOSTNAME, experiment, metric)
+
+
+def get_json_record(hostname, experiment, metric, timestamps, values):
+  """Creates a dict suitable for export to json.
+
+  Args:
+    hostname, str, hostname of host system.
+    experiment: str, name of experiment running on host.
+    metric: str, the canonical metric name for values.
+    timestamps: iterable of int, timestamps corresponding to each value.
+    values: iterable of float, values corresponding to each timestamp.
+  Returns:
+    dict, with keys for hostname, experiment, metric, and sample.
+  """
+  logging.debug('%s %s %s', hostname, experiment, metric)
+  json_data = {
+      'hostname': hostname, 'experiment': experiment, 'metric': metric}
+  json_data['sample'] = get_json_record_samples(timestamps, values)
+  return json_data
+
+
+def get_json_record_samples(timestamps, values):
+  """Converts a sequences of timestampes and values for a json record.
+
+  The timestamps and values arguments must be the same length.
+
+  Args:
+    timestamps: iterable of int, timestamps corresponding to each value.
+    values: iterable of float, values corresponding to each timestamp.
+  Returns:
+    list of dict, each dict has keys timestamp and value.
+  """
+  samples = []
+  assert(len(timestamps) == len(values))
+  for i in xrange(0, len(timestamps)):
+    if values[i] is not None:
+      samples.append({'timestamp': timestamps[i], 'value': values[i]})
+  return samples
+
+
+def write_json_record(fd_output, record, pretty_json):
+  """Writes json record to fd_output.
+
+  Args:
+    fd_output: file object open for writing, the record is written to this fd.
+    record: dict, the record of data to serialize as json.
+    pretty_json: bool, whether to write the json with extra spacing.
+  """
+  json.dump(record, fd_output, indent=pretty_json)
+  fd_output.write('\n')  # separate each record with newline.
+
+
+def get_rrd_files(rrddir_prefix):
+  """Returns the absolute path of all rrd files found under rrddir_prefix.
+
+  Args:
+    rrddir_prefix: str, base directory where rrd files are stored.
+  Returns:
+    list of str, where each element is the absolute path to a single rrd file.
+  """
+  rrdfiles = []
+  for root, _, filenames in os.walk(rrddir_prefix):
+    for filename in filenames:
+      file_ext = os.path.join(root, filename)
+      if file_ext.endswith('.rrd'):
+        rrdfiles.append(os.path.abspath(file_ext))
+  return rrdfiles
+
+
+def rrd_list(options):
+  """Processes all options.show_* flags without performing an export."""
+  for filename in get_rrd_files(options.rrddir_prefix):
+    (_, value_names, _) = rrdtool.fetch(
```

Comment:
So, no. I guess this was a style choice to group the LHS a little more, but I don't feel strongly about it. 9ad2648888affd1dbe048ad35047dbb9e3bb7590